services/management/adapter/file_provider.go

```diff
@@ -78,7 +83,8 @@ func (mp *FileProvider) generatePath(referenceTime primitives.TimestampSeconds)
 }
 
 func (mp *FileProvider) readUrl(path string) ([]byte, error) {
-	res, err := http.Get(path)
+	res, err := mp.client.Get(path)
```

Comment:
check err before doing defer or else res might not be defined

this is from the official docs:

```
resp, err := http.Get("http://example.com/")
if err != nil {
	// handle error
}
defer resp.Body.Close()
```